subsys/bluetooth/host/conn.c

```diff
@@ -2128,6 +2164,11 @@ void bt_conn_cb_register(struct bt_conn_cb *cb)
 	callback_list = cb;
 }
 
+void bt_conn_cb_unregister(void)
+{
+	memset(&callback_list, 0, sizeof(callback_list));
```

Comment:
`callback_list` is just a pointer.
```suggestion
	callback_list = NULL;
```

Or maybe go through the list and set the `_next` pointers to `NULL`.